src/util.ts

```diff
@@ -56,7 +56,13 @@ export class DeadlineRunner<T> {
       this._reject(error);
   }
 
+  private _isDebuggerAttached() {
+    return process.env.NODE_OPTIONS?.includes('ms-vscode.js-debug') || /--debug|--inspect/.test(process.execArgv.join(' '));
+  }
+
   setDeadline(deadline: number | undefined) {
+    if (this._isDebuggerAttached())
```

Comment:
I think this is not the best place, since it is a utility method. I'd rather make this work similarly to passing `--timeout=0` to the command line.